### ISSUE TITLE
bug: MD032 is not getting applied

### ISSUE THREAD

**Issue by martimlobao** [rvben/rumdl#220]
The bug seems to have appeared in v0.0.193.

```toml
[global]
disable = []
```

```md
# Heading

1. List
Some text.
```

```bash
❯ uv tool install rumdl==0.0.192
`rumdl==0.0.192` is already installed

❯ rumdl check test.md
test.md:3:1: [MD032] List should be followed by blank line [*]

Issues: Found 1 issues in 1 file (12ms)
Run `rumdl fmt` to automatically fix 1 of the 1 issues

❯ uv tool install rumdl==0.0.193
Resolved 1 package in 2ms
Uninstalled 1 package in 1ms
Installed 1 package in 2ms
 - rumdl==0.0.192
 + rumdl==0.0.193
Installed 1 executable: rumdl

❯ rumdl check test.md           

Success: No issues found in 1 file (27ms)
```

**Comment by rvben** [comment]
Thanks @martimlobao for raising this. The behavior change in v0.0.193 was intentional - MD032 now treats unindented text after lists as "lazy continuation" per CommonMark spec (matching markdownlint-cli).

For users who prefer the stricter behavior, I'll add this option in upcoming release:

```
[MD032]
allow-lazy-continuation = false
```

With this setting, your example will correctly warn about the missing blank line.
